sources/packages/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts

```diff
@@ -556,75 +557,79 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
   }
 
   /**
-   * Deny COE for an application.
-   ** Note: If an application has 2 COEs, and if the first COE is Rejected then 2nd COE is implicitly rejected.
-   * @param disbursementScheduleIds
-   * @param userId User who denies the COE.
-   * @param coeDeniedReasonId Denied reason id of a denied COE.
-   * @param otherReasonDesc If the denied reason is other, respective description.
+   * Deny COE for a disbursement schedule.
+   ** Note: If an application has 2 COEs, and if the first COE is rejected then 2nd COE is implicitly rejected.
+   * @param disbursementScheduleId disbursement schedule id to be updated.
+   * @param auditUserId user that should be considered the one that is causing the changes.
+   * @param coeDeniedReasonId denied reason id of a denied COE.
+   * @param otherReasonDesc if the denied reason is other, respective description.
    */
-  async updateCOEToDeny(
-    applicationId: number,
-    userId: number,
+  async updateCOEToDenied(
+    disbursementScheduleId: number,
+    auditUserId: number,
     coeDeniedReasonId: number,
     otherReasonDesc: string,
-  ): Promise<UpdateResult> {
-    return this.repo
-      .createQueryBuilder()
-      .update(DisbursementSchedule)
-      .set({
-        coeStatus: COEStatus.declined,
-        coeUpdatedBy: { id: userId },
-        coeUpdatedAt: new Date(),
-        coeDeniedReason: { id: coeDeniedReasonId },
-        coeDeniedOtherDesc:
-          coeDeniedReasonId === COE_DENIED_REASON_OTHER_ID
-            ? otherReasonDesc
-            : null,
-      })
-      .where("application.id = :applicationId", { applicationId })
-      .andWhere("coeStatus = :required", { required: COEStatus.required })
-      .execute();
+  ): Promise<DisbursementSchedule> {
+    const coeToUpdate = new DisbursementSchedule();
+    coeToUpdate.id = disbursementScheduleId;
```

Comment:
is this method safe to update a row? if by any chance the id is not present, it will create a new column with the id right? will affect the default id iteration of the db?. 

doesn't the below where condition required for updating?  `.where("application.id = :applicationId", { applicationId })
      .andWhere("coeStatus = :required", { required: COEStatus.required })`

let me know if I am missing anything

Reply:
A couple of notes.

About "will affect the default id iteration of the db?", typeorm prevents it. If you try to force it the column is decorated with the `PrimaryGeneratedColumn()` that will prevent it.

About "if by any chance the id is not present", the id is not received from the client and used without validation. At this point, the id is already validated and the method is called considering it. The service method can have more safeguards to ensure its own integrity, so I can change from a save to an update, for instance.

About "doesn't the below where condition required for updating?", I will add some of the conditions back but not in the same way. The update MUST NOT HAPPEN using the application id. This code is probably from the time when the COE was on the application.

Reply:
Method changed.